/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.userservice;

import "google/protobuf/empty.proto";
import public "models/user.proto";

service UserService {
  rpc UsersCreate (UsersCreateRequest) returns (User);

  rpc UsersDelete (UsersDeleteRequest) returns (google.protobuf.Empty);

  rpc UsersReadAll (google.protobuf.Empty) returns (UsersReadAllResponse);

  rpc UsersReadOne (UsersReadOneRequest) returns (User);

  rpc UsersUpdate (UsersUpdateRequest) returns (User);

}

message UsersCreateRequest {
  // User to create
  User user = 1;

}

message UsersDeleteRequest {
  // ID of the user.
  int32 userId = 1;

}

message UsersReadAllResponse {
  repeated User data = 1;
}

message UsersReadOneRequest {
  // ID of the user.
  int32 userId = 1;

}

message UsersUpdateRequest {
  // ID of the user.
  int32 userId = 1;
  User user = 2;

}

